### PR TITLE
Breathes new life into Venus Human Traps

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/49736
- https://github.com/tgstation/tgstation/pull/52650
- https://github.com/tgstation/tgstation/pull/53648
- https://github.com/tgstation/tgstation/pull/53880

## About The Pull Request

PR 1:
- Addresses venus human traps being able to damage you as fast as they can click (basicallly guaranteed death)
- The tethering/grappling function actually works. Prior to this, the vine beam attack would just damage you
- Vines stun carbons for 2 seconds when first applied, and will attempt to bring the affected closer to the plant. - If the target gets far enough away from the plant or shares the same tile as the plant, the vine will be destroyed.
- If a plant tries to attach a vine to something they already have a vine on, they will reel everything they can towards themselves. This also happens passively every second.
- Plants heal 10% of their max health when attacking with melee if their target is alive.
- Sentient plants can attach their vines to a wide variety of things, allowing them to take advantage of the damage dealt when walking though them.
- Plants can be possessed by ghosts akin to spidereggs

PR 2
- Adds a countdown to flowerbuds. For observers pleasure

PR 3
- Adds new sprites for venus human traps
- Replaces flower bud sprites with sprites that suck less
- They now have sounds for hitting, getting hit, and dying

PR 4 
- I fixed an old and broken CoffeeKat port (thanks as always Jupyter...)

MISC fixes
- VHTs no longer take cold damage (Not particularly fair, as Kudzu is predisposed to destroying glass)
- Fixes smoothing making the icon invisible
- fixes vines staying after the flower bulb is destroyed
- adds VHTs to organic biotype
- Gave VHTs unique_name = TRUE, so you can tell the difference between multiple
- Convert VHTs to use basepixels
- It now respects ghost pool protection

## Why It's Good For The Game
The look better, they perform better, they sound better, they dont *instantly* kill you

sounds like a win to me

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/62388554/99864d8b-8e96-47d5-a24a-2202f4d0b6ab

</details>

## Changelog
:cl: rkz, IndieanaJones, tralezab, Angustmeta
add: adds completely new sprites and sounds for Venus Human Traps!
fix: Cleans up VHTs to no longer instantly kill you due to spaghetti, and their tether knockdown ability actually works!
balance: Did I mention it no longer INSTANTLY KILLS YOU?
tweak: Observers now have a timer to which they can see when a new VHT will be born from a flower pod!
tweak: Observers can now control VHTs
fix: cleaned up some misc shitcode
/:cl:

### CODE REVIEW

**Review by Rukofamicom — APPROVED** [review]
Venus Man Trap code is broken to the point of bordering on non-functional, so this should not really be flagged as a balance change at all imo. The intended balance is still roughly the same as far as I can tell, they were not intended to instantly kill players. 

I plan to build on this in the near future as well.

**Comment by Fronsis** [comment]
Absolutely based, i've played a bit of flyhuman traps on TG and Skyrat and it works fine, the attack was a little clumsy until they fixed it but they were far from the instakill machine that they are now with the current code.

This certainly makes vine events more interesting and won't be welder spam until it's defeated(while they only broke one door) 

Does this port also uses the 2 insta humantraps that spawns when a vine event starts? Last time i played it worked like that, the vine generates and two humantraps controllable mob spawn to help defend it(to prevent the early kill)

File: code/modules/events/spacevine.dm [hunk 1]
```diff
@@ -561,8 +564,6 @@
 				master.spawn_spacevine_piece(stepturf, src)
 
 /obj/structure/spacevine/ex_act(severity, target)
-	if(istype(target, type)) //if its agressive spread vine dont do anything
-		return
```
Comment (Penwin0) [inline code/modules/events/spacevine.dm:565]: Is this meant to be removed?

File: code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm [hunk 2]
```diff
@@ -1,31 +1,73 @@
-/obj/structure/alien/resin/flower_bud_enemy //inheriting basic attack/damage stuff from alien structures
+
+#define FINAL_BUD_GROWTH_ICON 3
```
Comment (Penwin0) [inline code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm:2]: undefine and file end

File: code/modules/events/spacevine.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/modules/events/spacevine.dm:565]: Yup. Its part of a pr jupyterkat ported half of.

File: code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm (hunk 2 above)
Comment (Tsar-Salat) [reply code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm:2]: added in d009973

File: code/modules/events/spacevine.dm (hunk 1 above)
Comment (Penwin0) [reply code/modules/events/spacevine.dm:565]: Huh

**Comment by Rukofamicom** [comment]
Please do include the fact ghosts can now control VHT in the cl, that's fairly important for players who don't peruse github to know.

**Comment by PowerfulBacon** [comment]
Updated the PR body

**Comment by Tsar-Salat** [comment]
uh oh

**Pull request merged**